### ISSUE TITLE
Postgres Adapter (next/pg-adapter) doesn't satisfy Adapter type

### ISSUE THREAD

**Issue by chriddell** [nextauthjs/next-auth#11330]
### Adapter type

@auth/pg-adapter

### Environment

```
System:
    OS: macOS 14.5
    CPU: (10) arm64 Apple M1 Pro
    Memory: 98.58 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.19.0 - ~/.nvm/versions/node/v18.19.0/bin/node
    npm: 10.2.3 - ~/.nvm/versions/node/v18.19.0/bin/npm
  Browsers:
    Chrome: 126.0.6478.127
    Edge: 126.0.2592.87
    Safari: 17.5
  npmPackages:
    @auth/pg-adapter: ^1.4.1 => 1.4.1 
    next: ^13.4.8 => 13.4.8 
    next-auth: ^4.24.7 => 4.24.7 
    react: ^18.2.0 => 18.2.0
```

### Reproduction URL

https://github.com/chriddell/next-auth-pg-adapter-example

### Describe the issue

There seems to be an issue with the types of the Adapter exported from the @auth/pg-adapter package. I don't think they satisfy the type of Adapter specified by the next-auth package.

```ts
// /pages/api/auth/[...nextauth].ts

const options = {
  providers: [
    // list providers
  ],
  adapter: PostgresAdapter(pool),
};

// This next line throws a typescript error: 
export default Auth(options);
```

```
Argument of type '{ providers: OAuthConfig<AzureADProfile>[]; adapter: Adapter; }' is not assignable to parameter of type 'AuthOptions'.
```

### How to reproduce

1. Clone the repo linked
2. Open [...next-auth].ts
3. See the TS error on line 33

### Expected behavior

There is no Typescript error; PostgresAdapter satisfies Adapter.

**Comment by owlyowl** [comment]
Also seeing:
```
Adapter' is not assignable to type 'import("<......>/node_modules/next-auth/adapters").Adapter'.
  Types of property 'createUser' are incompatible.
    Type '((user: AdapterUser) => Awaitable<AdapterUser>) | undefined' is not assignable to type '((user: Omit<AdapterUser, "id">) => Awaitable<AdapterUser>) | undefined'.
      Type '(user: AdapterUser) => Awaitable<AdapterUser>' is not assignable to type '(user: Omit<AdapterUser, "id">) => Awaitable<AdapterUser>'.
        Types of parameters 'user' and 'user' are incompatible.
          Property 'id' is missing in type 'Omit<AdapterUser, "id">' but required in type 'AdapterUser'.ts(2322)
 ```
 
 *package.json*
 ```
  "dependencies": {
    "@auth/pg-adapter": "^1.5.0",
    "next": "14.2.13",
    "next-auth": "^4.24.7",
    "pg": "^8.13.0",
    "react": "^18",
    "react-dom": "^18"
  },
  "devDependencies": {
    "@types/node": "^20",
    "@types/pg": "^8.11.10",
    "@types/react": "^18",
    "@types/react-dom": "^18",
    "eslint": "^8",
    "eslint-config-next": "14.2.13",
    "postcss": "^8",
    "tailwindcss": "^3.4.1",
    "typescript": "^5"
  }
```

**Comment by balazsorban44** [comment]
This was recently fixed in 4.28.4 via https://github.com/nextauthjs/next-auth/pull/11562

Also sounds like a duplicate of #9493.

Please verify and open a new issue if the problem persists! Thanks!

**Comment by chriddell** [comment]
Confirmed this fixed the issue, thanks @balazsorban44
